src/Features/Core/Portable/ExtractInterface/ExtractInterfaceOptionsResult.cs

```diff
@@ -6,19 +6,27 @@ namespace Microsoft.CodeAnalysis.ExtractInterface
 {
     internal class ExtractInterfaceOptionsResult
     {
+        public enum ExtractLocation
+        {
+            SameFile,
+            NewFile
+        }
+
         public static readonly ExtractInterfaceOptionsResult Cancelled = new ExtractInterfaceOptionsResult(isCancelled: true);
 
         public bool IsCancelled { get; }
         public IEnumerable<ISymbol> IncludedMembers { get; }
```

Comment:
please make into an IMmutableArray.  It prevents accidentally doing things lazily, computing things too many times, or doing things on the wrong threads.